src/Microsoft.ML.CodeGenerator/CodeGenerator/CSharp/PipelineExtension.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.ML.AutoML;
+using Microsoft.ML.CodeGenerator.CSharp;
+
+namespace Microsoft.ML.CodeGenerator.CodeGenerator.CSharp
+{
+    internal static class PipelineExtension
+    {
+        internal static (string Usings, string TrainerMethod, List<string> PreTrainerTransforms, List<string> PostTrainerTransforms) GenerateTransformsAndTrainers(this Pipeline pipeline)
+        {
+            StringBuilder usingsBuilder = new StringBuilder();
+            var usings = new List<string>();
+
+            // Get pre-trainer transforms
+            var nodes = pipeline.Nodes.TakeWhile(t => t.NodeType == PipelineNodeType.Transform);
+            var preTrainerTransformsAndUsings = GenerateTransformsAndUsings(nodes);
+
+            // Get post trainer transforms
+            nodes = pipeline.Nodes.SkipWhile(t => t.NodeType == PipelineNodeType.Transform)
+                .SkipWhile(t => t.NodeType == PipelineNodeType.Trainer) //skip the trainer
+                .TakeWhile(t => t.NodeType == PipelineNodeType.Transform); //post trainer transforms
+            var postTrainerTransformsAndUsings = GenerateTransformsAndUsings(nodes);
+
+            //Get trainer code and its associated usings.
```

Comment:
May want to search for "//[a-zA-Z]" and add a space for the lines which are comments (and not commented out code).

```suggestion
            // Get trainer code and its associated usings.
```

Reply:
## Resolved